### ISSUE TITLE
Please add woff2 to fonts glob

### ISSUE THREAD

**Issue by alexciarlillo** [austinpray/asset-builder#14]
I am using font-awesome which now supports woff2. Modernizr detects a woff2-able browser and looks for the file but it is not being copied to the dist/ directory with the rest of the fonts. It looks like woff2 just needs to be added to types.json. I am sure there is a way to manually do this with the manifest file but I could not figure it out for the life of me. Could you show an example that might work? It may help me understand the asset-builder a bit better. Thanks

**Comment by austinpray** [comment]
Gotta do some work stuff real quick but this is a great opportunity for an example. Will fix and document in a moment.

**Comment by austinpray** [comment]
Alright so this should be fixed. 

Here is the commit: https://github.com/austinpray/asset-builder/commit/27dfc83bf0b238192b9b8bc27f5b29508e19c51f. I have added some tests to verify that this is working in the future.

I have also added some documentation about the default properties asset-builder sets for you: https://github.com/austinpray/asset-builder/commit/388122712764557a508613c48de43bfdd4c0e549

**There are three ways this could have been fixed.**

## 1. Patch asset-builder

### Background

Let me explain a bit how asset-builder works so the fix makes sense. Here is an example gulp task:

```js
gulp.src(manifest.globs.fonts, function () {
  // do the font stuff
});
```

The above is basically equivalent to this:

```js
gulp.src([
  'bower_components/example-font/example.woff', 
  'bower_components/example-font/example.woff2', 
  'assets/fonts/**/*'
], function () {
  // do the font stuff
});
```

Let's examine how `manifest.globs.fonts` gets built.

### Building `manifest.globs.fonts`

![asset-builder-flow](https://cloud.githubusercontent.com/assets/2192970/6035533/386d1468-abfb-11e4-962b-8d8151406087.png)


### Added some documentation

See https://github.com/austinpray/asset-builder/commit/27dfc83bf0b238192b9b8bc27f5b29508e19c51f

## Modify the gulpfile

Let's say there is a bug in asset builder such that no woff2 fonts are available in `manifest.globs.fonts`.

```js
console.log(manifest.globs.fonts);
// => ['bower_components/example-font/example.woff', 'assets/fonts/**/*']
```

There are a lot of ways to skin this cat. gulpfiles are just javascript. So you can do any array operation you want.

```js
gulp.src(manifest.globs.fonts.concat([
  'bower_components/example-font/example.woff2'
]), function () {
  // do the font stuff
});
```

or 

```js
manifest.globs.fonts.push('bower_components/example-font/example.woff2');
gulp.src(manifest.globs.fonts, function () {
  // do the font stuff
});
```

or you can just not even use asset-builder for that task

```js
gulp.src([
  'bower_components/example-font/example.woff', 
  'bower_components/example-font/example.woff2', 
  'assets/fonts/**/*'
], function () {
  // do the font stuff
});
```

gulpfiles are just javascript and asset-builder is just a node module that spits out arrays of globs.

## Modify the manifest.json

You could also have added the `bower_components/example-font/example.woff2` to the files property of the "fonts" dependency.

```json
{
  "dependencies": {
    "fonts": {
       "external": true,
       "files": [
         "assets/fonts/**/*",
         "bower_components/example-font/example.woff2"
       ]
    }
  }
}
```

**Comment by austinpray** [comment]
Run `npm update` to get the latest version of asset-builder for your project.

**Comment by alexciarlillo** [comment]
Thanks very much. I did get it working originally by modifying the gulpfile, but I knew the manifest file should be capable as well, and is where I would rather make modifications. I was close, but I was missing the "external" setting and putting "bower_components" in the path. Anyway, I really appreciate the explanation. The asset-builder makes a lot more sense to me now. I updated from npm and everything is good now.
